src/vcs/git.rs

```diff
@@ -192,6 +206,103 @@ impl<'repo> Repository {
         self.commit_to_history(commit)
     }
 
+    /// Get the [`Diff`] between two commits.
+    pub fn diff(&self, from: &'repo git2::Commit, to: &'repo git2::Commit) -> Result<Diff, Error> {
+        use git2::{Delta, Patch};
+
+        let mut diff = Diff::new();
+        let git_diff = self.diff_commits(from, Some(to))?;
+
+        for (idx, delta) in git_diff.deltas().enumerate() {
+            match delta.status() {
+                Delta::Added => {
+                    let diff_file = delta.new_file();
+                    let path = diff_file
+                        .path()
+                        .ok_or(Error::GitDiff("couldn't retrieve file path"))?;
+                    let path = file_system::Path::try_from(path.to_path_buf())?;
+
+                    diff.add_created_file(path);
+                },
+                Delta::Deleted => {
+                    let diff_file = delta.old_file();
+                    let path = diff_file
+                        .path()
+                        .ok_or(Error::GitDiff("couldn't retrieve file path"))?;
+                    let path = file_system::Path::try_from(path.to_path_buf())?;
+
+                    diff.add_deleted_file(path);
+                },
+                Delta::Modified => {
+                    let diff_file = delta.new_file();
+                    let path = diff_file
+                        .path()
+                        .ok_or(Error::GitDiff("couldn't retrieve file path"))?;
+                    let path = file_system::Path::try_from(path.to_path_buf())?;
+
+                    let patch = Patch::from_diff(&git_diff, idx)?;
+
+                    if let Some(patch) = patch {
+                        let mut hunks: Vec<Hunk> = Vec::new();
+
+                        for h in 0..patch.num_hunks() {
+                            let (hunk, hunk_lines) = patch.hunk(h)?;
+                            let header = hunk.header().to_owned();
```

Comment:
I was thinking, maybe we should have a type fo Header as well. What do you think? If we do want one then we can put it as a follow-up ticket -- doesn't need to be done now.

Reply:
Would just be an alias right? Not sure what these look like in pijul etc.

Reply:
That's what I'm unsure about. I'll create a follow-up ticket to figure exactly this out :)